### DISCUSSION TITLE
Clone a Repo Specified as Datasource to Specific Location

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [hairyhenderson/gomplate#1688] [Q&A]
Hey, is it possible to leverage the `git+https` datasource to clone a git repository to a specific location? I'd like the repository to include the `.git` folder so that I can commit from that folder as well.

**Comment by hairyhenderson** [comment]
Hi @ProfessorManhattan,

No, datasource functionality is purely read-only and clones are done in-memory only.

**Comment by ProfessorManhattan** [comment]
Hey @hairyhenderson -- would Go templating be able to unpack it? Something like:

```
for each PATH={{ ds "git" . | file.ReadDir }}
  {{ ds "git" "$PATH" }}
  {{ file.Read "$PATH" }}
end
```

I actually don't know. Was just wondering if it's something that someone else has done before.

**Reply by hairyhenderson** [reply]
> would Go templating be able to unpack it? Something like:

I'm not sure what you mean by "unpack"... What's your specific goal?

**Comment by ProfessorManhattan** [comment]
My goal is to only have got clone run once to reduce bandwidth requirements

**Reply by hairyhenderson** [reply]
Then you'll need to run `git clone` separately and use the file datasource instead.
